samples/net/mqtt_azure/prj.conf

```diff
@@ -0,0 +1,84 @@
+CONFIG_NETWORKING=y
+
+# Disable IPv6 support
+CONFIG_NET_IPV6=n
+
+# Enable IPv4 support
+CONFIG_NET_IPV4=y
+CONFIG_NET_IF_MAX_IPV4_COUNT=2
+CONFIG_NET_IF_UNICAST_IPV4_ADDR_COUNT=3
+
+# Enable TCP support
+CONFIG_NET_TCP=y
+
+# Enable Sockets support
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_SOCKOPT_TLS=y
+
+# Enable DHCPv4 support
+CONFIG_NET_DHCPV4=y
+
+# Enable SOCKS5 proxy support
+CONFIG_SOCKS=n
+
+# Enable MQTT Lib support
+CONFIG_MQTT_LIB=y
+CONFIG_MQTT_LIB_TLS=y
+
+# Enable Mbed TLS configuration
+CONFIG_MBEDTLS=y
+CONFIG_MBEDTLS_BUILTIN=y
+CONFIG_MBEDTLS_ENABLE_HEAP=y
+CONFIG_MBEDTLS_HEAP_SIZE=100000
+CONFIG_MBEDTLS_SSL_MAX_CONTENT_LEN=10240
+CONFIG_MBEDTLS_PEM_CERTIFICATE_FORMAT=y
+
+# Network configuration
+CONFIG_NET_CONFIG_SETTINGS=y
+
+# Network connection manager
+CONFIG_NET_CONNECTION_MANAGER=y
+
+CONFIG_NET_MGMT_EVENT_STACK_SIZE=1024
+CONFIG_NET_MGMT_EVENT_QUEUE_SIZE=5
+CONFIG_NET_MGMT_EVENT_LOG_LEVEL_DBG=n
+
+CONFIG_DNS_RESOLVER=y
+CONFIG_DNS_SERVER_IP_ADDRESSES=y
+CONFIG_DNS_SERVER1="8.8.8.8"
+CONFIG_DNS_RESOLVER_ADDITIONAL_BUF_CTR=2
+CONFIG_NEWLIB_LIBC=y
+
+CONFIG_INIT_STACKS=y
+CONFIG_NET_SHELL=y
+
+CONFIG_MAIN_STACK_SIZE=4096
+
+# Enable Logging support
+CONFIG_LOG_IMMEDIATE=y
+CONFIG_NET_LOG=y
+CONFIG_NET_TCP_LOG_LEVEL_DBG=n
+CONFIG_NET_SOCKETS_LOG_LEVEL_DBG=n
+CONFIG_MQTT_LOG_LEVEL_DBG=n
+CONFIG_NET_DHCPV4_LOG_LEVEL_INF=n
+CONFIG_NET_IF_LOG_LEVEL_DBG=n
+CONFIG_MBEDTLS_DEBUG=n
+#CONFIG_MBEDTLS_DEBUG_LEVEL=4
+CONFIG_SOCKS_LOG_LEVEL_DBG=n
+CONFIG_NET_CONFIG_LOG_LEVEL_DBG=y
+CONFIG_NET_CONNECTION_MANAGER_LOG_LEVEL_DBG=n
+CONFIG_DNS_RESOLVER_LOG_LEVEL_DBG=n
+CONFIG_NET_CONTEXT_LOG_LEVEL_DBG=n
+
+# Azure Cloud User Settings
```

Comment:
Perhaps the Azure specific credentials / configuration could be placed to overlay file, then end user would not need to edit this generic prj.conf file. This would mean some tweaks to the readme file to document this.